daliuge-translator/dlg/dropmake/dm_utils.py

```diff
@@ -533,30 +500,21 @@ def convert_construct(lgo):
             app_node["group"] = k_new
             app_node["group_start"] = 1
 
-            # extra step to deal with "internal output" fromo within Gather
+            # extra step to deal with "internal output" from within Gather
             dup_app_node_k = min(keyset) - 1
             keyset.add(dup_app_node_k)
-            dup_app_node = dict()
-            dup_app_node["key"] = dup_app_node_k
-            dup_app_node["category"] = node[has_app]  # node['application']
-            dup_app_node["name"] = (
-                node["text"] if "text" in node else node["name"]
-            )
-
-            if "mkn" in node:
-                dup_app_node["mkn"] = node["mkn"]
+            dup_app_args = {
+                "key": dup_app_node_k,
+                "fields": "appFields" if "appFields" in node else "inputAppFields"
+            }
+            tmp_node = _create_from_node(node, node[has_app], dup_app_args)
```

Comment:
in the original code there was also a "category" field. Is that not required anymore?

Reply:
It's still there - `up_app_node["category"]` is  `node[has_app]`, which is the second parameter to the `_create_from_node()` method (`_create_from_node(node: dict, category: str, app_params: dict) -> dict:`).

I'll use the arguments explicitly in the function call to make it clearer.